File: src/scrapers/test_scraper.py
#!/usr/bin/env python3
import os
from residential_scraper import scrape_residential_listings

# Test the scraper with a real API key if available
api_key = os.environ.get('SCRAPE_DO_API_KEY', 'test_key_here')

if api_key == 'test_key_here':
    print("⚠️  No SCRAPE_DO_API_KEY found in environment variables")
    print("Set it with: export SCRAPE_DO_API_KEY='your_actual_key'")
else:
    url = "https://www.vivanuncios.com.mx/s-renta-inmuebles/distrito-federal/v1c30l1029p1"
    print(f"Testing scraper with URL: {url}")
    
    results = scrape_residential_listings(url, api_key)
    
    if results:
        print(f"\n✅ Found {len(results)} listings:")
        for i, listing in enumerate(results[:3], 1):  # Show first 3
            print(f"\n{i}. {listing.get('title', 'No title')}")
            print(f"   Price: {listing.get('price', 'N/A')}")
            print(f"   Location: {listing.get('location', 'N/A')}")
            print(f"   URL: {listing.get('url', 'N/A')}")
    else:
        print("\n❌ No listings found. Check if selectors need updating.")

File: src/scrapers/testScraper.js
import { scrapeResidentialListings } from './residentialScraper.js';

async function test() {
  const apiKey = process.env.SCRAPE_DO_API_KEY;
  
  if (!apiKey || apiKey === 'your_api_key_here') {
    console.log('⚠️  No SCRAPE_DO_API_KEY found in environment variables');
    console.log('Set it with: export SCRAPE_DO_API_KEY="your_actual_key"');
    return;
  }
  
  const url = 'https://www.vivanuncios.com.mx/s-renta-inmuebles/distrito-federal/v1c30l1029p1';
  console.log(`Testing scraper with URL: ${url}\n`);
  
  const results = await scrapeResidentialListings(url, apiKey);
  
  if (results.length > 0) {
    console.log(`✅ Found ${results.length} listings:\n`);
    
    // Show first 3 listings
    results.slice(0, 3).forEach((listing, index) => {
      console.log(`${index + 1}. ${listing.title || 'No title'}`);
      console.log(`   Price: ${listing.price || 'N/A'}`);
      console.log(`   Location: ${listing.location || 'N/A'}`);
      console.log(`   URL: ${listing.url || 'N/A'}\n`);
    });
  } else {
    console.log('❌ No listings found. Check if selectors need updating.');
  }
}

test();